torch/csrc/jit/codegen/cuda/graph_fuser.cpp

```diff
@@ -1276,6 +1292,48 @@ void PeepholeOptimizeShapeExpressions(Block* block) {
   }
 }
 
+Value* guardView(
+    Node* fusion,
+    std::unordered_map<Value*, Value*>& valueToSize,
+    Node* versioning_if,
+    Node* view,
+    Value* view_sizes_profile_ivalue) {
+  // 1. Get self tensor sizes and view_sizes
+  auto self = view->inputs().front();
+  auto self_type = self->type()->cast<TensorType>();
+  auto self_sizes = getTensorSizes(self_type);
+  auto self_shape = valueToSize[self];
+  TORCH_INTERNAL_ASSERT(valueToSize.find(self) != valueToSize.end());
```

Comment:
:red_square: We are checking after accessing. That mean we'll never assert here and this can be a surprise later.